keras_cv/backend/config.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import os
+
+_MULTI_BACKEND = False
+
+# Set Keras base dir path given KERAS_HOME env variable, if applicable.
+# Otherwise either ~/.keras or /tmp.
+if "KERAS_HOME" in os.environ:
+    _keras_dir = os.environ.get("KERAS_HOME")
+else:
+    _keras_base_dir = os.path.expanduser("~")
+    if not os.access(_keras_base_dir, os.W_OK):
+        _keras_base_dir = "/tmp"
+    _keras_dir = os.path.join(_keras_base_dir, ".keras")
+
+# Attempt to read KerasCV config file.
+_config_path = os.path.expanduser(os.path.join(_keras_dir, "keras-cv.json"))
+if os.path.exists(_config_path):
+    try:
+        with open(_config_path) as f:
+            _config = json.load(f)
+    except ValueError:
+        _config = {}
+    _MULTI_BACKEND = _config.get("multi_backend", _MULTI_BACKEND)
+
+# Save config file, if possible.
+if not os.path.exists(_keras_dir):
+    try:
+        os.makedirs(_keras_dir)
+    except OSError:
+        # Except permission denied and potential race conditions
+        # in multi-threaded environments.
+        pass
+
+if not os.path.exists(_config_path):
+    _config = {
+        "multi_backend": _MULTI_BACKEND,
+    }
+    try:
+        with open(_config_path, "w") as f:
+            f.write(json.dumps(_config, indent=4))
+    except IOError:
+        # Except permission denied.
+        pass
+
+# Set Keras backend based on KERAS_CV_MULTI_BACKEND flag, if applicable.
+if "KERAS_CV_MULTI_BACKEND" in os.environ:
```

Comment:
@fchollet

I think both me and @ianstenbit were thinking it would be way less user friction if in `KERAS_BACKEND=xx` just worked, without the need for this mouthful every time. Something like...

Transient flow:
- `KERAS_BACKEND=xx python cv-script.py` just works.
- `KERAS_BACKEND=tensorflow python cv-script.py` is keras-core with tf.
- `python cv-script.py` is `tf.keras`.

Persistent flow:
- Edit `keras.json` to set `backend: "xx"`.
- Edit `keras_cv.json` to set `multi_backend: True`

Reply:
I've updated the code to reflect this approach.